app/uk/gov/hmrc/digitalservicestax/journeys/RegJourney.scala

```diff
@@ -26,47 +26,41 @@ import cats.implicits._
 import java.time.LocalDate
 
 import uk.gov.hmrc.digitalservicestax.frontend._
-import ltbs.uniform.{NonEmptyString => _, _}
+import ltbs.uniform._
 import ltbs.uniform.validation._
 import ltbs.uniform.validation.Rule._
+import izumi.reflect.TagK
 
 object RegJourney {
 
-  type RegTellTypes = Confirmation[Registration] :: CYA[Registration] :: Address :: Kickout :: Company :: Boolean :: NilTypes
-  type RegAskTypes = UTR :: Postcode :: LocalDate :: ContactDetails :: UkAddress :: ForeignAddress :: Boolean :: CompanyName :: NilTypes
-
-
   private def message(key: String, args: String*) = {
     import play.twirl.api.HtmlFormat.escape
     Map(key -> Tuple2(key, args.toList.map { escape(_).toString } ))
   }
 
   def nameCheck(name: CompanyName): Boolean = name.toString.matches("""^[a-zA-Z0-9 '&-]{1,105}$""")
 
-  def registrationJourney[F[_] : Monad](
-    interpreter: Language[F, RegTellTypes, RegAskTypes],
+  def registrationJourney[F[_] : Monad : TagK](
     backendService: DSTService[F]
-  ): F[Registration] = {
-    import interpreter._
+  ) = {
 
     for {
       globalRevenues <- ask[Boolean]("global-revenues")
-      _ <- if (!globalRevenues) { end("global-revenues-not-eligible", Kickout("global-revenues-not-eligible")) } else { (()).pure[F] }
+      _ <- end("global-revenues-not-eligible", Kickout("global-revenues-not-eligible")) unless globalRevenues
```

Comment:
I think you should be able to do this in a single 'step' - 

```
_ <- end("global-revenues-not-eligible", Kickout("global-revenues-not-eligible")) unless ask[Boolean]("global-revenues")
```

you may prefer to keep it as two steps anyway, there's no performance advantage either way. 

you may also want to consider doing away with the Kickout class altogether - 

```
_ <- end("global-revenues-not-eligible") unless ask[Boolean]("global-revenues")
```